activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -705,7 +713,16 @@ def unique_constraints(table_name)
           scope = quoted_scope(table_name)
 
           unique_info = internal_exec_query(<<~SQL, "SCHEMA", allow_retry: true, materialize_transactions: false)
-            SELECT c.conname, c.conrelid, c.conkey, c.condeferrable, c.condeferred, pg_get_constraintdef(c.oid) AS constraintdef
+            SELECT c.conname, c.conrelid, c.condeferrable, c.condeferred, pg_get_constraintdef(c.oid) AS constraintdef,
```

Comment:
Unique constraints is a very rarely used feature (compared to indexes and foreign keys) and the table usually have at most one of them. This query is now much more complicated and I think not worth it, so I would avoid doing it.

Reply:
I agree that unique constraints are used less-frequently than indexes/FKs. As one data point, at work we do at least make use of them in ~10 cases out of ~150 tables, and of those cases, it turns out most are composite constraints. In terms of additional complication, I personally think it's ok, but it's not my call 😄 A few points:
1. It's the same sub-query pattern as in `foreign_keys`, so there's a single additional "concept" to understand. (It would have been the same pattern in `indexes` too, before your PR showed us `pg_get_indexdef` for one `indkey` position - it's a shame there isn't something similar for FKs/constraints!)
3. It allows us to remove `column_names_from_column_numbers` (and avoid the associated trip to/from the DB), which prevents it being used when inappapropriate in the future
4. In the future, if/when the minimum supported version of PG is increased, we can reduce some of the incidental complexity:

```sql
(
  SELECT array_agg(a.attname ORDER BY idx)
  FROM (
    SELECT idx, c.conkey[idx] AS conkey_elem
    FROM generate_subscripts(c.conkey, 1) AS idx
  ) indexed_conkeys
  JOIN pg_attribute a ON a.attrelid = t.oid
  AND a.attnum = indexed_conkeys.conkey_elem
) AS conkey_names
```
can become
```sql
(
  SELECT array_agg(a.attname ORDER BY idx)
  FROM unnest(c.conkey) WITH ORDINALITY i(conkey_elem, idx)
  JOIN pg_attribute a ON a.attrelid = t.oid 
  AND a.attnum = i.conkey_elem
) AS conkey_names
```

The main point of my original PR was `indexes`, which your PR improved. Therefore I feel less strongly about the remainder in this PR, but I thought it was good to make them consistently avoid separate queries. If you or others feel strongly, I'm fine with removing this change for `unique_constraints`.